File: buff/classes/354/pub/sum/c-static/sum.c
// C sum program

#include <stdio.h>

static int sum(int seq[], int n) {
  int s=0;
  for (int i=0; i<n; i++)
    s+=seq[i];
  return s;
}

int main() {
  static int seq[]={5,6,1,8,3,7}; // static allocation
  printf("%d\n",sum(seq,sizeof(seq)/sizeof(int)));
  return 0;
}

File: buff/classes/354/pub/etc/Sum.java
// Java OO sum program

public class Sum {

    private int s;

    public Sum() {
        init();
    }

    public void init() {
        s = 0;
    }

    public int sum(int[] seq) {
        for (int v : seq)
            s += v;
        return s;
    }

    public static void main(String[] args) {
        int[] seq = { 5, 6, 1, 8, 3, 7 };
        Sum sum = new Sum();
        System.out.println(sum.sum(seq));
    }

}
